retention/cohort_analysis.py

```diff
@@ -157,8 +159,24 @@ def user_agent_to_client(user_agent):
     return OTHER
 
 
-# select all users that created an account for a given range
-def get_new_users(start, stop):
+def get_new_users(start: int, stop: int) -> Sequence[Tuple[str, str]]:
+    """Get a list of all users that registered an account during
+    the given timeframe
+
+    Also gets the IDs of any devices that they used during that period.
+
+    Args:
+        start: start of the timeframe to check, inclusive, as ms since the
+            epoch.
+
+        stop: end of the timeframe to check, exclusive, as ms since the
+            epoch.
+    Returns:
+        A sequence of (user id, device id) pairs.
+    """
+
+    # XXX why do we check user_daily_visits here, particularly given we
```

Comment:
As to why we're joining on `user_daily_visits` at all:
* There appear to be some `user` rows that don't have `device_id`s in `user_daily_visits` and I think we want to exclude them (they never completed registration?).
* I'm unsure about the `user_daily_visits` timestamp check. I suspect that excludes users who started registration on one day but didn't visit until a later day (is that possible?). Unsure if that is desired, would ask @neilisfragile given that restriction existed prior to my changes - https://github.com/vector-im/stats/pull/2/commits/4eb7e692f718ec6972bf3edf59a968729c930af7#diff-f0f0f09e7fa1012c80dc4ce6487dad25c51e495960343d91d92078effa92dda6L139

As to why `device_id` is being returned, I agree it doesn't appear functionally necessary. I do wonder whether we wanted it for the logging to compare the number of user-devices to the number of user-devices-user-agents returned from `get_user_agents`. But it can probably be removed